src/email/services/email.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@nestjs/common';
+import { Email } from '@prisma/client';
+
+import { NodeMailerService } from './node-mailer.service';
+import { EmailRepository } from '../../database/repositories/email.repository';
+import { IExchangeRate } from '../../rate/interfaces';
+import { SubscribeEmailDto } from '../dtos/subscribe-email.dto';
+import { AlreadySubscribedException } from '../exceptions';
+import { ISendCurrentRateContext } from '../interfaces/send-email-options.interface';
+
+@Injectable()
+export class EmailService {
+  constructor(
+    private readonly emailRepository: EmailRepository,
+    private readonly nodeMailerService: NodeMailerService,
```

Comment:
What will you do here if the requirements change and you need to use for example MailChimp client for email sending or other one, not NodeMailer ?

Reply:
If I need to add one more client for sending email, I would use Strategy pattern to use different implementation of email sending logic. If I need to replace the node mailer then I would either rewrite the service that is responsible for sending email or create a new one. And I think that name nodeMailerService can be replaced with more common name like MailingService.